JASP-Engine/JASP/R/selectionmodels.R

```diff
@@ -939,36 +989,137 @@ jaspResults[["plot_estimates"]] <- plot_estimates
+    fit[["adj_est"]][posMean, 1]     <- fit_old[["adj_est"]][posMean, 1]     * -1
+    fit[["z_adj"]][posMean, 1]       <- fit_old[["z_adj"]][posMean, 1]       * -1
+    fit[["ci.lb_unadj"]][posMean, 1] <- fit_old[["ci.ub_unadj"]][posMean, 1] * -1
+    fit[["ci.ub_unadj"]][posMean, 1] <- fit_old[["ci.lb_unadj"]][posMean, 1] * -1
+    
+    fit[["output_unadj"]][["par"]][posMean] <- fit_old[["output_unadj"]][["par"]][posMean] * -1    
+    fit[["output_adj"]][["par"]][posMean]   <- fit_old[["output_adj"]][["par"]][posMean]   * -1
+    
+    fit[["estimates_turned"]] <- TRUE
+  }
+  
+  return(fit)
+}
 
-    fit[["output_unadj"]][["par"]][mean_pos] <- fit_old[["output_unadj"]][["par"]][mean_pos] * -1    
-    fit[["output_adj"]][["par"]][mean_pos]   <- fit_old[["output_adj"]][["par"]][mean_pos]   * -1
+.smTransformEstimates      <- function(fit, options) {
   
-    fit[["estimates_turned"]] <- TRUE
+  if (options[["measures"]] == "general")
+    return(fit)
+  
+  # in the case that correlation input was used, this part will transform the results back
+  # from the estimation scale to the outcome scale
+  
+  if (!class(fit) %in% c("simpleError","error")) {
+    
+    if (fit[["fe"]]) {
+      posMean <- 1    
+    } else {
+      posMean <- 2
+    }
+    
+    fit_old <- fit
+    
+    
+    fit[["unadj_est"]][posMean, 1] <- .smInvTransform(fit_old[["unadj_est"]][posMean, 1], options[["mu_transform"]])
+    fit[["ci.lb_adj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.lb_adj"]][posMean, 1], options[["mu_transform"]])
+    fit[["ci.ub_adj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.ub_adj"]][posMean, 1], options[["mu_transform"]])
+    
+    fit[["adj_est"]][posMean, 1]     <- .smInvTransform(fit_old[["adj_est"]][posMean, 1],     options[["mu_transform"]])
+    fit[["ci.lb_unadj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.lb_unadj"]][posMean, 1], options[["mu_transform"]])
+    fit[["ci.ub_unadj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.ub_unadj"]][posMean, 1], options[["mu_transform"]])
+    
+    fit[["output_unadj"]][["par"]][posMean] <- .smInvTransform(fit_old[["output_unadj"]][["par"]][posMean], options[["mu_transform"]])
+    fit[["output_adj"]][["par"]][posMean]   <- .smInvTransform(fit_old[["output_adj"]][["par"]][posMean],   options[["mu_transform"]])
+    
+    fit[["estimates_transformed"]] <- options[["mu_transform"]]
   }
   
   return(fit)
 }
+
+.smGetInputES              <- function(dataset, options) {
+  
+  # change the direction
+  ES <- dataset[, .v(options[["input_ES"]])] * ifelse(options[["effect_direction"]] == "negative", -1, 1)
+  
+  # do a scale transform if neccessary
+  if (options[["measures"]] == "general") {
+    return(ES)
+  } else if (options[["measures"]] == "correlation") {
+    return(.smTransform(ES, transformation = options[["mu_transform"]], what = "ES"))
+  }
+}
+
+.smGetInputVAR             <- function(dataset, options) {
```

Comment:
What's VAR?

Reply:
shortcut for variance

Reply:
Then `.smGetInputVar` would do I reckon.

Reply:
in the same way as ES stands for Effect Size

Reply:
so it's VARiance then?

Reply:
I renamed it to `.smGetInputVar`